### PR TITLE
CDAP-12577 look in spark directory of system artifact directories

### PR DESCRIPTION
**Pull request by albertshau**
This is for setups where users have set the system artifact
directory to some non-default value. When they upgrade, they will
likely miss the spark specific system artifacts located in the
artifacts directory.

### CODE REVIEW

**Comment by anwar6953** [comment]
Let's also revert the cdap-default.xml value back to its previous value?

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-RUT1381

File: cdap-common/src/main/resources/cdap-default.xml [hunk 1]
```diff
@@ -315,7 +315,7 @@
 
   <property>
     <name>app.artifact.dir</name>
-    <value>/opt/cdap/master/artifacts;/opt/cdap/master/artifacts/${app.program.spark.compat}</value>
+    <value>/opt/cdap/master/artifacts</value>
```
Comment (anwar6953) [inline cdap-common/src/main/resources/cdap-default.xml:318]: also update `cdap-standalone/src/main/resources/cdap-site.xml`

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/DefaultArtifactRepository.java [hunk 2]
```diff
@@ -97,16 +98,24 @@ public DefaultArtifactRepository(CConfiguration cConf, ArtifactStore artifactSto
...
+      String sparkDirStr = SparkCompat.get(cConf).getCompat();
+
       for (String dir : systemArtifactsDir.split(";")) {
         File file = new File(dir);
         if (!file.isDirectory()) {
           LOG.warn("Ignoring {} because it is not a directory.", file);
           continue;
         }
         systemArtifactDirs.add(file);
+        // if users have a non-default value for system artifacts, they may be missing the spark compat directory
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/DefaultArtifactRepository.java]: Even the non-default value for system artifacts needs this now (not just non-default values).
So this comment is somewhat inaccurate / misleading.

**Review by anwar6953 — APPROVED** [review]

**Pull request merged**